src/Dapr.AspNetCore/DaprServiceCollectionExtensions.cs

```diff
@@ -32,16 +32,41 @@ public static class DaprServiceCollectionExtensions
     /// </summary>
     /// <param name="services">The <see cref="IServiceCollection" />.</param>
     /// <param name="configure"></param>
-    public static void AddDaprClient(this IServiceCollection services, Action<DaprClientBuilder>? configure = null)
+    /// <param name="lifetime">The lifetime of the registered services.</param>
+    public static void AddDaprClient(this IServiceCollection services, Action<DaprClientBuilder>? configure = null,
+        ServiceLifetime lifetime = ServiceLifetime.Singleton)
     {
         ArgumentNullException.ThrowIfNull(services, nameof(services));
 
-        services.TryAddSingleton(serviceProvider =>
+        switch (lifetime)
         {
-            var builder = CreateDaprClientBuilder(serviceProvider);
-            configure?.Invoke(builder);
-            return builder.Build();
-        });
+
+            case ServiceLifetime.Scoped:
+                services.TryAddScoped(serviceProvider =>
+                {
+                    var builder = CreateDaprClientBuilder(serviceProvider);
+                    configure?.Invoke(builder);
+                    return builder.Build();
+                });
```

Comment:
This lambda could be reused across each lifetime option?

Reply:
Good call